lib/ex_doc/autolink.ex

```diff
@@ -116,8 +116,8 @@ defmodule ExDoc.Autolink do
         fragment = (uri.fragment && "#" <> uri.fragment) || ""
         HTML.text_to_id(without_ext) <> config.ext <> fragment
       else
-        message = "documentation references file `#{uri.path}` but it doesn't exist"
-        warn(message, config.file, config.line, config.id)
+        maybe_warn(nil, config, nil, %{file_path: uri.path, original_text: href})
```

Comment:
warning for missing extras is quite different from all the other ones: there's a file_path & original_text, there's neither ref nor visibility, should we call warn/3 directly (basically as before) to remove this indirection?

Reply:
Nope. Because maybe_warn checks the files that are excluded in the config, such as `skip_undefined_reference_warnings_on` and the upcoming  `skip_undefined_reference_warnings_for`